### DISCUSSION TITLE
Tamil models on spaCy

### DISCUSSION THREAD

**Discussion by koaning** [explosion/spaCy#10412] [Help: Coding & Implementations]
I'm starting a thread here so that [a discussion](https://twitter.com/Aswin01238441/status/1498952107063275522?s=20&t=qM-fpSe5n26UUWeM3Bc7kA) moves from Twitter to Github. 

It seems that the goal is to train a Tamil model so that others may use it. I do not speak Tamil, but since spaCy does have a tokenizer for it (see the [language guide](https://spacy.io/usage/models#languages)) I'd argue that nothing is stopping anyone from making their own model. One might even go as far as pre-populating some vectors. Though I cannot gauge the quality; both [fasttext](https://fasttext.cc/docs/en/crawl-vectors.html) and [bpemb](https://bpemb.h-its.org/) seem to support it. 

The goal is to have this be a place for discussions on how to train a Tamil model. I personally think it’s better to worry about having a representative dataset first. It's very hard to make a meaningful general model so I'm inclined to first focus on a specific corpus that might yield a model that can be re-used. This is a personal gut feeling and I do not speak Tamil, though so I'll gladly hear it if other folks have suggestions.

**Comment by AswiN-7** [comment]
dataset: 
https://github.com/UniversalDependencies/UD_Tamil-TTB

code done:
simple tokenizer, lemmatizer, pos tagger
https://github.com/AswiN-7/tamilNLP 

question:
how to add this things spacy model

**Reply by adrianeboyd** [reply]
Hi, a good example for setting up a spacy pipeline for a UD corpus is here: https://github.com/explosion/projects/tree/v3/pipelines/tagger_parser_ud

```bash
python -m spacy project clone pipelines/tagger_parser_ud
```

For working with spacy projects: https://spacy.io/usage/projects

Be aware that this particular corpus is relatively small, so you may not see very good results, especially for the parser.

**Reply by AswiN-7** [reply]
hi
 

         python -m spacy pipelines/tagger_parser_ud

using the above command i cloned the project and  for downloading assets changed values in project.yml  file

![Screenshot 2022-03-09 210419](https://user-images.githubusercontent.com/60695838/157474710-a423f94b-c87a-4f18-b20f-b0791cb5af3c.jpg)

         python -m spacy project assets

after running the above command assets are downloaded 

![Screenshot 2022-03-09 205750](https://user-images.githubusercontent.com/60695838/157476576-7251b397-197f-434d-860f-79cde47cea46.jpg)

along with this

![Screenshot 2022-03-09 205632](https://user-images.githubusercontent.com/60695838/157476741-5554e55a-e3ab-4643-a80f-1c796541bf5a.jpg)

this error is coming and i cant able to run 

       python -m spacy run preprocess

Anybody know what to do about this??

**Reply by polm** [reply]
Please don't share screenshots of code or errors or file trees, just copy and paste text or terminal output. It is very hard to read your screenshots.

It looks like you just have a permissions error. Maybe there is a better way to solve it on Windows but I think it should go away if you run as Administrator.

Note that many of the example projects have only been tested on Linux/OSX and may have issues on Windows.

**Reply by AswiN-7** [reply]
now i'm shifted to ubuntu

           python -m spacy project assets

           python3 -m spacy project run preprocess

           ✘ Missing dependency specified by command 'preprocess': assets/UD_Tamil-TTB/ta_ttb-ud--test.conllu
           Maybe you forgot to run the 'project assets' command or a previous step?

what to do?

**Reply by polm** [reply]
You should be using the same `python` throughout your code, whether that's `python` or `python3`. (I'm not sure if those are the same on Ubuntu.)

Does the indicated filename exist? Do you not have a typo in the filename, like with the `--` part?
